metadata-ingestion/src/datahub/ingestion/source/datahub/datahub_database_reader.py

```diff
@@ -101,13 +118,20 @@ def query(self) -> str:
         # Relies on createdon order to reflect version order
         # Ordering of entries with the same createdon is handled by VersionOrderer
         return f"""
-            SELECT urn, aspect, metadata, systemmetadata, createdon, version
-            FROM {self.engine.dialect.identifier_preparer.quote(self.config.database_table_name)}
-            WHERE createdon >= %(since_createdon)s
-            {"" if self.config.include_all_versions else "AND version = 0"}
+            SELECT * from (
+            SELECT CONCAT(mav.createdon, "-", mav.urn, "-",  mav.aspect, "-", mav.version) as row_id ,mav.urn, mav.aspect, mav.metadata, mav.systemmetadata, mav.createdon, mav.version, removed
+            FROM {self.engine.dialect.identifier_preparer.quote(self.config.database_table_name)} as mav
+            LEFT JOIN (SELECT *, JSON_EXTRACT(metadata, '$.removed') as removed from  {self.engine.dialect.identifier_preparer.quote(self.config.database_table_name)} where aspect = "status" and version = 0) as sd on sd.urn = mav.urn
+            WHERE 1 = 1
+            {"" if self.config.include_all_versions else "AND mav.version = 0"}
+            {"" if not self.config.exclude_aspects else f"AND LOWER(mav.aspect) NOT IN %(exclude_aspects)s"}
+            AND mav.createdon >= %(since_createdon)s
+            ORDER BY createdon, urn, aspect, version
```

Comment:
can we indent the subquery here - otherwise the sql is hard to read

also - I like how we format sql queries in the fivetran source https://github.com/datahub-project/datahub/blob/a11ac8d104649c953037d4b85afac67d6828ad18/metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran_query.py#L134 - I'd like to move towards this type of formatting everywhere